frontend/model/contracts/group.js

```diff
@@ -936,22 +930,14 @@ sbp('chelonia/defineContract', {
                   // If already joined, ignore this error
                   if (e?.name === 'GIErrorUIRuntimeError' && e.cause?.name === 'GIGroupAlreadyJoinedError') return
                   console.error('Error while joining the #General chatroom', e)
-                  const errMsg = L("Couldn't join the #{chatroomName} in the group. An error occurred: {error}", { chatroomName: CHATROOM_GENERAL_NAME, error: e?.message || e })
-                  const promptOptions = {
-                    heading: L('Error while joining a chatroom'),
-                    question: errMsg,
-                    primaryButton: L('Close')
-                  }
-
-                  sbp('gi.ui/prompt', promptOptions)
+
+                  // TODO: event
                 })
               }
             } else {
-              // avoid blocking the main thread
-              // eslint-disable-next-line require-await
-              (async () => {
-                alert(L("Couldn't join the #{chatroomName} in the group. Doesn't exist.", { chatroomName: CHATROOM_GENERAL_NAME }))
-              })()
+              console.error('Couldn\'t join the chatroom in the group. Doesn\'t exist.', { chatroomName: CHATROOM_GENERAL_NAME })
+
+              // TODO: event
```

Comment:
Two things:

1. Minor readability nit: could you use `"` instead of `'` for the outer quotes so that you don't have to write `\'`?
2. What are these `// TODO: event` comments? Could you please either do the TODO or edit it so that it elaborates in detail what needs to be done and why? (I certainly don't know what this means, and likely future-you will forget too)

Reply:
If you see the code that's been removed, it was informing the user of an issue there. That's not really possible with the SW (and even then, it'd lead to convoluted code), so this is meant to say that an event should be fired. I'll add the events.